request-blocklist/tests.json

```diff
@@ -0,0 +1,352 @@
+        "requestUrl": "https://request-blocklist.example/sometimes-block",
+        "requestType": "image",
+        "websiteUrl": "https://another.subdomain.different.example",
+        "expectAction": "block",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "matching rule, non-matching domain",
+        "requestUrl": "https://request-blocklist.example/sometimes-block",
+        "requestType": "image",
+        "websiteUrl": "https://other-website.example",
+        "expectAction": "allow",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "matching rule, non-matching domain",
+        "requestUrl": "https://request-blocklist.example/sometimes-block",
+        "requestType": "image",
+        "websiteUrl": "https://different.example",
+        "expectAction": "allow",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "non-matching rule",
+        "requestUrl": "https://other-website.example/no-match",
+        "requestType": "image",
+        "websiteUrl": "https://website.example",
+        "expectAction": "allow",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "matching rule with wildcard, <all> domains",
+        "requestUrl": "https://request-blocklist.example/path/block",
+        "requestType": "image",
+        "websiteUrl": "https://website.example",
+        "expectAction": "block",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "matching rule, port in URL",
+        "requestUrl": "https://request-blocklist.example:3000/port-block",
+        "requestType": "image",
+        "websiteUrl": "https://website.example",
+        "expectAction": "block",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "non-matching rule due to port",
+        "requestUrl": "https://request-blocklist.example/port-block",
+        "requestType": "image",
+        "websiteUrl": "https://website.example",
+        "expectAction": "allow",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "matching rule, first-party request",
+        "requestUrl": "https://request-blocklist.example/block",
+        "requestType": "image",
+        "websiteUrl": "https://request-blocklist.example",
+        "expectAction": "block",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "matching rule, query part of URL considered",
+        "requestUrl": "https://request-blocklist.example/incorrect?query-part",
+        "requestType": "image",
+        "websiteUrl": "https://website.example",
+        "expectAction": "block",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "matching rule, '.' literal considered",
+        "requestUrl": "https://request-blocklist.example/.path/regex",
+        "requestType": "image",
+        "websiteUrl": "https://website.example",
+        "expectAction": "block",
+        "exceptPlatforms": []
+      },
+      {
+        "name": "matching rule, case considered",
```

Comment:
@dharb So far I went with case-sensitive matching. My thinking there is that case-sensitive matching is both more efficient + more powerful. But that is at odds with the Tracker Allowlist which I think goes with case-insensitive matching. I'm open to either, WDYT?